src/data_hub/_api.py

```diff
@@ -182,18 +180,38 @@ def dynamic_function(
                 title="Deep Origin error: [Name conflicts]",
             )
 
-        if debug:
-            print(f"Response request = {response.http_request}")
-            print("Response headers:")
-            _print_dict(dict(response.headers), json=False)
-            print(f"Response = {response.http_response}")
+        if not isinstance(response, dict):
             response = response.json()
 
-        if hasattr(response, "data"):
-            return response.data
+        if "data" in response.keys():
+            response = response["data"]
```

Comment:
Worth thinking about how to deal with paginated requests, which will include paginated information at the top-level of `response`. Food for thought.